server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -161,39 +172,25 @@ public void getSynonymRules(String resourceName, int from, int size, ActionListe
             .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, resourceName))
             .setFrom(from)
             .setSize(size)
+            .addSort("id", SortOrder.ASC)
             .setPreference(Preference.LOCAL.type())
             .setTrackTotalHits(true)
-            .execute(new ActionListener<>() {
-                @Override
-                public void onResponse(SearchResponse searchResponse) {
-                    final long totalSynonymRules = searchResponse.getHits().getTotalHits().value;
-                    if (totalSynonymRules == 0) {
-                        listener.onFailure(new ResourceNotFoundException("Synonym set [" + resourceName + "] not found"));
-                        return;
-                    }
-                    final SynonymRule[] synonymRules = Arrays.stream(searchResponse.getHits().getHits())
-                        .map(SynonymsManagementAPIService::hitToSynonymRule)
-                        .toArray(SynonymRule[]::new);
-                    listener.onResponse(new PagedResult<>(totalSynonymRules, synonymRules));
-                }
-
-                @Override
-                public void onFailure(Exception e) {
-                    final Throwable cause = ExceptionsHelper.unwrapCause(e);
-                    if (cause instanceof IndexNotFoundException) {
-                        listener.onFailure(new ResourceNotFoundException("Synonym set [" + resourceName + "] not found"));
-                        return;
-                    }
-                    listener.onFailure(e);
+            .execute(new DelegatingIndexNotFoundActionListener<>(resourceName, listener, (l, searchResponse) -> {
+                final long totalSynonymRules = searchResponse.getHits().getTotalHits().value;
+                if (totalSynonymRules == 0) {
+                    l.onFailure(new ResourceNotFoundException("Synonym set [" + resourceName + "] not found"));
+                    return;
                 }
-            });
+                final SynonymRule[] synonymRules = Arrays.stream(searchResponse.getHits().getHits())
+                    .map(SynonymsManagementAPIService::hitToSynonymRule)
+                    .toArray(SynonymRule[]::new);
+                l.onResponse(new PagedResult<>(totalSynonymRules, synonymRules));
+            }));
     }
 
     private static SynonymRule hitToSynonymRule(SearchHit hit) {
-        return new SynonymRule(
-            externalSynonymRuleId(hit.getId()),
-            (String) hit.getSourceAsMap().get(SynonymRule.SYNONYMS_FIELD.getPreferredName())
-        );
+        Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+        return new SynonymRule((String) sourceAsMap.get(SYNONYM_RULE_ID_FIELD), (String) sourceAsMap.get(SYNONYMS_FIELD));
```

Comment:
Do we need to use `externalSynonymRuleId` function?  
May be we can add another test to ensure proper ids.

Reply:
> Do we need to use externalSynonymRuleId function?

Now we can remove it, thanks!

> May be we can add another test to ensure proper ids.

What do you have in mind?

Reply:
About the test I was thinking to have a check on synonyms IDs that they don't change (but I see that is already covered by our current yml tests). So no action needed.